src/mysql_conn.erl

```diff
@@ -616,13 +618,18 @@ named_prepare(Name, Query, State) ->
 
 %% @doc Transforms result sets into a structure appropriate to be returned
 %% to the client.
-handle_query_call_result([], _Query, State, []) ->
+handle_query_call_result([_] = Recs, Query, State) ->
+    handle_query_call_result(Recs, single_result, 1, Query, State, []);
+handle_query_call_result(Recs, Query, State) ->
+    handle_query_call_result(Recs, multi_result, 1, Query, State, []).
+
+handle_query_call_result([], _Type, _N, _Query, State, []) ->
     {ok, State};
-handle_query_call_result([], _Query, State, [{ColumnNames, Rows}]) ->
+handle_query_call_result([], _Type, _N, _Query, State, [{ColumnNames, Rows}]) ->
     {{ok, ColumnNames, Rows}, State};
-handle_query_call_result([], _Query, State, ResultSetsAcc) ->
+handle_query_call_result([], _Type, _N, _Query, State, ResultSetsAcc) ->
     {{ok, lists:reverse(ResultSetsAcc)}, State};
-handle_query_call_result([Rec|Recs], Query, State = #state{transaction_levels = L}, ResultSetsAcc) ->
+handle_query_call_result([Rec|Recs], Type, N, Query, State = #state{transaction_levels = L}, ResultSetsAcc) ->
```

Comment:
Maybe `Type` could have a more descriptive name, e.g. `MultiResult`. A type makes me think of things like integer, boolean and such things (i.e. datatypes).

Also line break to 80, preferably.

Reply:
`Type` was removed by implementing above suggestion. Reduced line length to <80 as well.

Reply:
`Type` -> obsoleted by implementing above suggestion.

Line wdth =<80 -> done